renpy/loadsave.py

```diff
@@ -409,25 +411,15 @@ def save(slotname, extra_info='', mutate_flag=False, include_screenshot=True):
     logf = io.BytesIO()
     try:
         dump((roots, renpy.game.log), logf)
-    except Exception:
-
-        t, e, tb = sys.exc_info()
-
-        if mutate_flag:
-            reraise(t, e, tb)
-
-        try:
-            bad = find_bad_reduction(roots, renpy.game.log)
-        except Exception:
-            reraise(t, e, tb)
-
-        if bad is None:
-            reraise(t, e, tb)
-
-        if e.args:
-            e.args = (e.args[0] + ' (perhaps {})'.format(bad),) + e.args[1:]
+    except Exception as e:
+        if not mutate_flag and e.args:
+            try:
+                bad = find_bad_reduction(roots, renpy.game.log)
+                e.args = (e.args[0] + f' (perhaps {bad})', *e.args[1:])
```

Comment:
I think you'd need a conditional here to get equivalence with the original, otherwise you could get "perhaps (None)":
```suggestion
                if bad := find_bad_reduction(roots, renpy.game.log):
                    e.args = (e.args[0] + f' (perhaps {bad})', *e.args[1:])
```